### PR TITLE
Add a system/resource channel to share system resources like system images

### PR DESCRIPTION
**Pull request by xster**
Engine half of https://github.com/flutter/flutter/issues/60034 / https://github.com/flutter/flutter/issues/60034

doc: go/cupertino-icons-next-step (sorry Google internal)

Alternatives considered: adding the new 'SystemImage.load' to 'flutter/platform' channel. But 'flutter/platform' is a JSON codec'ed channel and changing it will be a breaking change. I also couldn't migrate Fuchsia's 'flutter/platform' to standard codec.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterResourcePlugin.mm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        case 700:
+          symbolWeight = UIImageSymbolWeightBold;
+          break;
+        case 800:
+          symbolWeight = UIImageSymbolWeightHeavy;
+          break;
+        case 900:
+          symbolWeight = UIImageSymbolWeightBlack;
+          break;
+      }
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterResourcePlugin.mm:59]: Probably should have a default case that returns the error state.

File: shell/platform/darwin/ios/framework/Source/FlutterResourcePlugin.mm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+#include <Foundation/Foundation.h>
+#include <UIKit/UIKit.h>
+
+@implementation FlutterResourcePlugin
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  id args = call.arguments;
+  if ([method isEqualToString:@"SystemImage.load"]) {
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterResourcePlugin.mm:16]: Please pull this string literal into a variable so that the test can reference it.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 3]
```diff
@@ -235,7 +238,7 @@ - (void)setViewController:(FlutterViewController*)viewController {
   _viewController =
       viewController ? [viewController getWeakPtr] : fml::WeakPtr<FlutterViewController>();
   self.iosPlatformView->SetOwnerViewController(_viewController);
-  [self maybeSetupPlatformViewChannels];
+  [self maybeSetupChannelsHandler];
```
Comment (gaaclarke) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:241]: Might be worth adding a unit test to FlutterEngine to make sure resourceChannel isn't nil after setting the view controller.

**Comment by xster** [comment]
Anyway, this turned out to be a big saga but I'm going a different route now. https://github.com/flutter/flutter/pull/65083 is the new way.
